assert_screenshots.js

```diff
@@ -0,0 +1,91 @@
+const fs = require('fs');
+const {PNG} = require('pngjs');
+const pixelmatch = require('pixelmatch');
+const mkdirpCb = require('mkdirp');
+const {promisify} = require('util');
+const path = require('path');
+const assert = require('assert');
+
+const mkdirp = promisify(mkdirpCb);
+
+/**
+ * Create an image showing the difference between image A and B.
+ * @param {import('pngjs').PNGWithMetadata} a 
+ * @param {import('pngjs').PNGWithMetadata} b 
+ * @returns {{diff_image: import('pngjs').PNG, diff_pixel_count: number}}
+ */
+function create_diff_image(a, b) {
+    // Get max dimensions
+    const width = Math.max(a.width, b.width);
+    const height = Math.max(a.height, b.height);
+
+    const diff = new PNG({width, height});
+    const diff_pixel_count = pixelmatch(
+        a.data,
+        b.data,
+        diff.data,
+        width,
+        height,
+        {threshold: 0.1}
+    );
+
+    return {
+        diff_image: diff,
+        diff_pixel_count
+    };
+}
+
+/**
+ * Take a screenshot of the current page and compare it with the
+ * previous one if any exists.
+ * @param {import('puppeteer').Page} page 
+ * @param {*} config 
+ * @param {string} name 
+ */
+async function assert_screenshot(page, config, name) {
+    assert(name);
+
+    await mkdirp(config.screenshot_directory);
+
+    const actual = path.join(
+        config.screenshot_directory,
+        `${config.task_name}-${name}-actual.png`
+    );
+    const expected = path.join(
+        config.screenshot_directory,
+        `${config.task_name}-${name}-expected.png`
+    );
+
+    const has_expected = fs.existsSync(expected);
```

Comment:
Why do we use synchronous code here and below? Can't we `promisify` any non-async methods?